### PR TITLE
[new release] vchan, vchan-xen and vchan-unix (6.0.1)

### PR DESCRIPTION
**Pull request by dinosaure**
Xen Vchan implementation

- Project page: <a href="https://github.com/mirage/ocaml-vchan">https://github.com/mirage/ocaml-vchan</a>
- Documentation: <a href="https://mirage.github.io/ocaml-vchan">https://mirage.github.io/ocaml-vchan</a>

##### CHANGES:

* Adapt to mirage-xen.7.0.0 API changes (mirage/ocaml-vchan#138 @dinosaure)

### CODE REVIEW

**Comment by mseri** [comment]
Lower bounds on cstruct and fmt are needed. I will try to have a look tomorrow, if you already know what they are please update the opam files

File: packages/vchan/vchan.6.0.1/opam [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+license: "ISC"
+tags: "org:mirage"
+homepage: "https://github.com/mirage/ocaml-vchan"
+doc: "https://mirage.github.io/ocaml-vchan"
+bug-reports: "https://github.com/mirage/ocaml-vchan/issues"
+depends: [
+  "ocaml" {>= "4.08.0"}
+  "dune"
+  "lwt" {>= "2.5.0"}
+  "cstruct" {>= "1.9.0"}
```
Comment (mseri) [inline packages/vchan/vchan.6.0.1/opam:19]:
```
```suggestion
  "cstruct" {>= "6.0.0"}
```
For
```
#=== ERROR while compiling vchan.6.0.1 ========================================#
# context              2.1.2 | linux/x86_64 | ocaml-base-compiler.4.08.1 | pinned(https://github.com/mirage/ocaml-vchan/releases/download/v6.0.1/vchan-6.0.1.tbz)
# path                 ~/.opam/4.08/.opam-switch/build/vchan.6.0.1
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p vchan -j 31
# exit-code            1
# env-file             ~/.opam/log/vchan-8363-99c8eb.env
# output-file          ~/.opam/log/vchan-8363-99c8eb.out
### output ###
#       ocamlc lib/.vchan.objs/vchan__Endpoint.{cmo,cmt} (exit 2)
# (cd _build/default && /home/opam/.opam/4.08/bin/ocamlc.opt -w -40 -g -bin-annot -I lib/.vchan.objs -I lib/.vchan.objs/.private -I /home/opam/.opam/4.08/lib/base/caml -I /home/opam/.opam/4.08/lib/bytes -I /home/opam/.opam/4.08/lib/cstruct -I /home/opam/.opam/4.08/lib/fmt -I /home/opam/.opam/4.08/lib/io-page -I /home/opam/.opam/4.08/lib/lwt -I /home/opam/.opam/4.08/lib/mirage-flow -I /home/opam/.opam/4.08/lib/parsexp -I /home/opam/.opam/4.08/lib/ppx_sexp_conv/runtime-lib -I /home/opam/.opam/4.08/lib/result -I /home/opam/.opam/4.08/lib/seq -I /home/opam/.opam/4.08/lib/sexplib -I /home/opam/.opam/4.08/lib/sexplib0 -I /home/opam/.opam/4.08/lib/xenstore -I /home/opam/.opam/4.08/lib/xenstore/client -intf-suffix .ml -no-alias-deps -open Vchan -o lib/.vchan.objs/vchan__Endpoint.cmo -c -impl lib/endpoint.pp.ml)
# File "lib/endpoint.ml", line 282, characters 12-26:
# 282 |   let len = Cstruct.length buf in
#                   ^^^^^^^^^^^^^^
# Error: Unbound value Cstruct.length
#     ocamlopt lib/.vchan.objs/vchan__Endpoint.{cmx,o} (exit 2)
# (cd _build/default && /home/opam/.opam/4.08/bin/ocamlopt.opt -w -40 -g -I lib/.vchan.objs -I lib/.vchan.objs/.private -I /home/opam/.opam/4.08/lib/base/caml -I /home/opam/.opam/4.08/lib/bytes -I /home/opam/.opam/4.08/lib/cstruct -I /home/opam/.opam/4.08/lib/fmt -I /home/opam/.opam/4.08/lib/io-page -I /home/opam/.opam/4.08/lib/lwt -I /home/opam/.opam/4.08/lib/mirage-flow -I /home/opam/.opam/4.08/lib/parsexp -I /home/opam/.opam/4.08/lib/ppx_sexp_conv/runtime-lib -I /home/opam/.opam/4.08/lib/result -I /home/opam/.opam/4.08/lib/seq -I /home/opam/.opam/4.08/lib/sexplib -I /home/opam/.opam/4.08/lib/sexplib0 -I /home/opam/.opam/4.08/lib/xenstore -I /home/opam/.opam/4.08/lib/xenstore/client -intf-suffix .ml -no-alias-deps -open Vchan -o lib/.vchan.objs/vchan__Endpoint.cmx -c -impl lib/endpoint.pp.ml)
# File "lib/endpoint.ml", line 282, characters 12-26:
# 282 |   let len = Cstruct.length buf in
#                   ^^^^^^^^^^^^^^
# Error: Unbound value Cstruct.length
```

File: packages/vchan-unix/vchan-unix.6.0.1/opam [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  "io-page"
+  "mirage-flow" {>= "2.0.0"}
+  "xenstore" {>= "1.2.2"}
+  "xenstore_transport" {>= "1.0.0"}
+  "xen-evtchn-unix"
+  "xen-gnt-unix"
+  "sexplib"
+  "xen-evtchn"
+  "xen-gnt"
+  "cmdliner" {with-test}
```
Comment (mseri) [inline packages/vchan-unix/vchan-unix.6.0.1/opam:27]:
```suggestion
  "fmt" {>= "0.8.7"}
  "cmdliner" {with-test}
```
For
```

#=== ERROR while compiling vchan-unix.6.0.1 ===================================#
# context              2.1.2 | linux/x86_64 | ocaml-base-compiler.4.13.1 | pinned(https://github.com/mirage/ocaml-vchan/releases/download/v6.0.1/vchan-6.0.1.tbz)
# path                 ~/.opam/4.13/.opam-switch/build/vchan-unix.6.0.1
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p vchan-unix -j 31
# exit-code            1
# env-file             ~/.opam/log/vchan-unix-10037-563148.env
# output-file          ~/.opam/log/vchan-unix-10037-563148.out
### output ###
[...]
#       ocamlc lwt/.vchan_unix.objs/byte/vchan_lwt_unix.{cmo,cmt} (exit 2)
# (cd _build/default && /home/opam/.opam/4.13/bin/ocamlc.opt -w -40 -g -bin-annot -I lwt/.vchan_unix.objs/byte -I /home/opam/.opam/4.13/lib/base/caml -I /home/opam/.opam/4.13/lib/bigarray-compat -I /home/opam/.opam/4.13/lib/bytes -I /home/opam/.opam/4.13/lib/cstruct -I /home/opam/.opam/4.13/lib/fmt -I /home/opam/.opam/4.13/lib/io-page -I /home/opam/.opam/4.13/lib/io-page-unix -I /home/opam/.opam/4.13/lib/lwt -I /home/opam/.opam/4.13/lib/lwt/unix -I /home/opam/.opam/4.13/lib/mirage-flow -I /home/opam/.opam/4.13/lib/mirage-profile -I /home/opam/.opam/4.13/lib/mmap -I /home/opam/.opam/4.13/lib/ocaml/threads -I /home/opam/.opam/4.13/lib/ocplib-endian -I /home/opam/.opam/4.13/lib/parsexp -I /home/opam/.opam/4.13/lib/ppx_sexp_conv/runtime-lib -I /home/opam/.opam/4.13/lib/result -I /home/opam/.opam/4.13/lib/seq -I /home/opam/.opam/4.13/lib/sexplib -I /home/opam/.opam/4.13/lib/sexplib0 -I /home/opam/.opam/4.13/lib/stdlib-shims -I /home/opam/.opam/4.13/lib/vchan -I /home/opam/.opam/4.13/lib/xen-evtchn -I /home/opam/.opam/4.13/lib/xen-evtchn-unix -I /home/opam/.opam/4.13/lib/xen-gnt -I /home/opam/.opam/4.13/lib/xen-gnt-unix -I /home/opam/.opam/4.13/lib/xenstore -I /home/opam/.opam/4.13/lib/xenstore/client -I /home/opam/.opam/4.13/lib/xenstore_transport -intf-suffix .ml -no-alias-deps -o lwt/.vchan_unix.objs/byte/vchan_lwt_unix.cmo -c -impl lwt/vchan_lwt_unix.pp.ml)
# File "lwt/vchan_lwt_unix.ml", line 88, characters 21-29:
# 88 |         | Error e -> Fmt.kstr Lwt.fail_with  "%a" M.pp_error e
#                           ^^^^^^^^
# Error: Unbound value Fmt.kstr
# Hint: Did you mean kstrf?
#     ocamlopt lwt/.vchan_unix.objs/native/vchan_lwt_unix.{cmx,o} (exit 2)
# (cd _build/default && /home/opam/.opam/4.13/bin/ocamlopt.opt -w -40 -g -I lwt/.vchan_unix.objs/byte -I lwt/.vchan_unix.objs/native -I /home/opam/.opam/4.13/lib/base/caml -I /home/opam/.opam/4.13/lib/bigarray-compat -I /home/opam/.opam/4.13/lib/bytes -I /home/opam/.opam/4.13/lib/cstruct -I /home/opam/.opam/4.13/lib/fmt -I /home/opam/.opam/4.13/lib/io-page -I /home/opam/.opam/4.13/lib/io-page-unix -I /home/opam/.opam/4.13/lib/lwt -I /home/opam/.opam/4.13/lib/lwt/unix -I /home/opam/.opam/4.13/lib/mirage-flow -I /home/opam/.opam/4.13/lib/mirage-profile -I /home/opam/.opam/4.13/lib/mmap -I /home/opam/.opam/4.13/lib/ocaml/threads -I /home/opam/.opam/4.13/lib/ocplib-endian -I /home/opam/.opam/4.13/lib/parsexp -I /home/opam/.opam/4.13/lib/ppx_sexp_conv/runtime-lib -I /home/opam/.opam/4.13/lib/result -I /home/opam/.opam/4.13/lib/seq -I /home/opam/.opam/4.13/lib/sexplib -I /home/opam/.opam/4.13/lib/sexplib0 -I /home/opam/.opam/4.13/lib/stdlib-shims -I /home/opam/.opam/4.13/lib/vchan -I /home/opam/.opam/4.13/lib/xen-evtchn -I /home/opam/.opam/4.13/lib/xen-evtchn-unix -I /home/opam/.opam/4.13/lib/xen-gnt -I /home/opam/.opam/4.13/lib/xen-gnt-unix -I /home/opam/.opam/4.13/lib/xenstore -I /home/opam/.opam/4.13/lib/xenstore/client -I /home/opam/.opam/4.13/lib/xenstore_transport -intf-suffix .ml -no-alias-deps -o lwt/.vchan_unix.objs/native/vchan_lwt_unix.cmx -c -impl lwt/vchan_lwt_unix.pp.ml)
# File "lwt/vchan_lwt_unix.ml", line 88, characters 21-29:
# 88 |         | Error e -> Fmt.kstr Lwt.fail_with  "%a" M.pp_error e
#                           ^^^^^^^^
# Error: Unbound value Fmt.kstr
# Hint: Did you mean kstrf?
```

File: packages/vchan-unix/vchan-unix.6.0.1/opam [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  "xen-gnt-unix"
+  "sexplib"
+  "xen-evtchn"
+  "xen-gnt"
+  "fmt" {>= "0.8.7"}
+  "cmdliner" {with-test}
+  "ounit" {with-test}
+]
+build: [
+  ["dune" "subst"] {pinned}
```
Comment (kit-ty-kate) [inline packages/vchan-unix/vchan-unix.6.0.1/opam:32]:
```suggestion
  ["dune" "subst"] {dev}
```

File: packages/vchan-xen/vchan-xen.6.0.1/opam [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  "mirage-flow" {>= "2.0.0"}
+  "xenstore" {>= "1.2.2"}
+  "mirage-xen" {>= "7.0.0"}
+  "xenstore_transport" {>= "1.0.0"}
+  "sexplib"
+  "ounit" {with-test}
+  "cmdliner" {with-test}
+]
+build: [
+  ["dune" "subst"] {pinned}
```
Comment (kit-ty-kate) [inline packages/vchan-xen/vchan-xen.6.0.1/opam:28]:
```suggestion
  ["dune" "subst"] {dev}
```

File: packages/vchan/vchan.6.0.1/opam [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+  "io-page"
+  "mirage-flow" {>= "2.0.0"}
+  "xenstore" {>= "1.2.2"}
+  "xenstore_transport" {>= "1.0.0"}
+  "sexplib"
+  "cmdliner" {with-test}
+  "ounit" {with-test}
+]
+build: [
+  ["dune" "subst"] {pinned}
```
Comment (kit-ty-kate) [inline packages/vchan/vchan.6.0.1/opam:31]:
```suggestion
  ["dune" "subst"] {dev}
```

**Comment by kit-ty-kate** [comment]
Thanks

**Pull request merged**